Source/Extensions/Blazorise.Markdown/Markdown.razor.cs

```diff
@@ -272,20 +272,34 @@ public Task NotifyCustomButtonClicked( string name, object value )
     /// <summary>
     /// Notifies the component that file input value has changed. This method is intended for internal framework use only and should not be called directly by user code.
     /// </summary>
-    /// <param name="file">Changed file.</param>
+    /// <param name="fileEntries">Array of changed file</param>
     /// <returns>A task that represents the asynchronous operation.</returns>
     [JSInvokable]
-    public async Task NotifyImageUpload( FileEntry file )
+    public async Task NotifyImageUpload( FileEntry[] fileEntries )
     {
-        file.FileUploadEndedCallback = new();
+        foreach ( var file in fileEntries )
+        {
+            file.FileUploadEndedCallback = new();
 
-        // So that method invocations on the file can be dispatched back here
-        file.Owner = (IFileEntryOwner)(object)this;
+            // So that method invocations on the file can be dispatched back here
+            file.Owner = this;
+        }
 
         if ( ImageUploadChanged is not null )
-            await ImageUploadChanged.Invoke( new( file ) );
+        {
+            await ImageUploadChanged.Invoke( new( fileEntries ) );
+        }
+
+        foreach ( var file in fileEntries )
+        {
+            file.FileUploadEndedCallback.SetResult();
+        }
+
+        foreach ( var file in fileEntries )
```

Comment:
This seems to be working fine on my side. Not sure if there will be any sideefects now with your solution so please check.

Also, the finalizer was never reached for me in debug mode. Maybe we don't need it anymore?

Reply:
It is working, until user code hits the https://github.com/Megabit/Blazorise/issues/5977 sub-issue 2. If we can live with that than the finalizer isn't needed...

Reply:
We never meant to support that anyway. The file should be handled asap during the changed event.

Reply:
Yes, that's why I suggested the rename and the api change to `TryWriteToStreamAsync`, so users don't shoot their own leg with our "shoulds".

Reply:
Nah :)

OK, please remove finalizers from final versions. And add release notes before I merge it.

Reply:
Ok.

You mean release notes here https://github.com/Megabit/Blazorise/issues/5857 or creating a release note 175 .razor?

Reply:
> It is working, until user code hits the #5977 sub-issue 2. If we can live with that than the finalizer isn't needed...

I should have done more tests. It seems we have now broken FilePicker. And FIlePicker works by saving references for later upload. So, I think we will need to go back to the finalizer or find an alternative for 

```
foreach ( var file in files )
{
    await file.Owner.RemoveFileEntry( file.Id );
}
```